packages/rule-components/src/RuleTable/RuleTable.js

```diff
@@ -0,0 +1,227 @@
+                ]
+            });
+        } else {
+            const currIndex = expanded.findIndex(id => id === (ruleId || key));
+            expanded.splice(currIndex, 1);
+            this.setState({ expanded });
+        }
+    }
+
+    onFilterChange = (type, _e, value) => {
+        this.setState({
+            filterValues: {
+                ...this.state.filterValues,
+                [type]: value
+            }
+        }, () => this.onUpdate(true));
+    }
+
+    onSort = (_e, index, direction) => {
+        const { fetchData, rules, columns, detail } = this.props;
+        const { filterValues } = this.state;
+        fetchData({
+            meta: calculateMeta(rules.meta),
+            filterValues,
+            sortBy: { index, direction, column: columns[index - Boolean(detail)] }
+        });
+    }
+
+    onPaginate = (perPage, page) => {
+        const { fetchData, rules, columns, detail, sortBy } = this.props;
+        const { filterValues } = this.state;
+        fetchData({
+            meta: { ...calculateMeta(rules.meta), perPage, page },
+            filterValues,
+            sortBy: { ...sortBy, column: columns[sortBy.index - Boolean(detail)] }
+        });
+    }
+
+    render() {
+        const {
+            rules,
+            filters,
+            columns,
+            detail,
+            actions,
+            ariaLabel,
+            fetchData,
+            isLoading,
+            sortBy,
+            toolbarProps,
+            ...props
+        } = this.props;
+        const { expanded, filterValues } = this.state;
+        const { meta, data } = rules;
+
+        const rows = flatten(createRows(data, columns, expanded, detail));
+        const defaultMeta = calculateMeta(meta);
+        const pagination = {
+            ...defaultMeta,
+            onPerPageSelect: (_e, perPage) => this.onPaginate(perPage, 1),
+            onSetPage: (_e, page) => this.onPaginate(defaultMeta.perPage, page)
+        };
+        const filterItems = Object.entries(filters).map(([ key, filter ]) => ({
+            ...filter({
+                onChange: (...props) => this.onFilterChange(key, ...props),
+                value: filterValues[key],
+                filterKey: key
+            })
+        }));
+
+        return <section className="rhcs-c-rules-table">
+            <PrimaryToolbar
+                filterConfig={{ items: filterItems }}
+                activeFiltersConfig={{
+                    filters: Object.entries(filterValues).map(([ key, value ]) => calculateActiveFilters(
+                        filterItems.find(({ filterKey }) => filterKey === key),
+                        value,
+                        key
+                    )).filter(Boolean),
+                    onDelete: this.onFilterDelete
+                }}
+                pagination={pagination}
+                { ...toolbarProps }
+            />
+            <Table
+                {...props}
+                {...!isLoading && { actions } }
```

Comment:
I suggest to avoid this contstructions as well, because it's really hard to read, it can be easily transformed into:
`actions={!isLoading && actions}`

Reply:
Patternfly table has problems with some attributes even if they have value set to `false` I usually try to play safe and do this instead of assign with check. But looks like the actions can be false and plays nicely, I'll change it to suggested approach.